engine/src/main/scala/cromwell/engine/workflow/WorkflowStoreActor.scala

```diff
@@ -78,37 +78,35 @@ abstract class WorkflowStore {
   }
 
   def remove(id: WorkflowId): Boolean = {
-    val newWorkflowStore = workflowStore filterNot { _.id == id }
-
-    if (newWorkflowStore == workflowStore) {
-      false
-    } else {
-      workflowStore = newWorkflowStore
+    if (workflowStore.exists(_.id == id)) {
+      workflowStore = workflowStore filterNot { _.id == id }
       true
+    } else {
+      false
     }
   }
 }
 
-class WorkflowStoreActor extends WorkflowStore with Actor with ServiceRegistryClient {
+class WorkflowStoreActor extends WorkflowStore with Actor with ActorLogging with ServiceRegistryClient {
   /*
     WARNING: WorkflowStore is NOT thread safe. Unless that statement is no longer true do NOT use threads
     outside of the the single threaded Actor event loop
    */
 
-  private val logger = Logging(context.system, this)
-
   override def receive = {
     case SubmitWorkflow(source) =>
       val id = add(NonEmptyList(source)).head
       registerIdWithMetadataService(id)
       sender ! WorkflowSubmittedToStore(id)
+      log.info(s"Workflow $id submitted.")
```

Comment:
Compared to other things we have I find this pretty useful though. Without this there's no way to know when looking at the logs if the workflow was submitted except waiting for it to be picked up by the next sweep (at best).